corehq/apps/userreports/views.py

```diff
@@ -1110,6 +1110,37 @@ class CreateDataSourceView(BaseEditDataSourceView):
     page_title = ugettext_lazy("Create Data Source")
 
 
+class CloneDataSourceView(CreateDataSourceView):
+    urlname = 'clone_configurable_data_source'
+
+    @property
+    def config_id_to_clone(self):
+        return self.kwargs.get('config_id_to_clone')
+
+    def page_url(self):
+        return reverse(self.urlname, kwargs=dict(domain=self.domain, config_id_to_clone=self.config_id_to_clone))
+
+    @property
+    def config(self):
+        data_source_to_clone = get_datasource_config_or_404(self.config_id_to_clone, self.domain)[0]
+        return DataSourceConfiguration(
+            domain=self.domain,
+            referenced_doc_type=data_source_to_clone.referenced_doc_type,
+            table_id=data_source_to_clone.table_id,
```

Comment:
will the new data source point to the same table?  If so that seems problematic.